### PR TITLE
CDPSDX-3528; fix configuration conditions to enable new sdx optimization config values

### PR DESCRIPTION
**Pull request by RokolyA94**
CDPSDX-3528; fix configuration conditions to enable new sdx optimization config values.

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java [hunk 1]
```diff
@@ -59,39 +59,45 @@ public class CoreConfigProvider extends AbstractRoleConfigProvider {
...
     }
 
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
         List<ApiClusterTemplateConfig> apiClusterTemplateConfigs = new ArrayList<>();
-        Optional<ApiClusterTemplateConfig> roleConfig = templateProcessor.getRoleConfig(CORE_SETTINGS, STORAGEOPERATIONS, CORE_DEFAULTFS);
-        if (roleConfig.isEmpty()) {
-            ConfigUtils.getStorageLocationForServiceProperty(source, CORE_DEFAULTFS)
-                    .ifPresent(location -> apiClusterTemplateConfigs.add(config(CORE_DEFAULTFS, location.getValue())));
+        if (!templateProcessor.isRoleTypePresentInService(HDFS, Lists.newArrayList(NAMENODE))
```
Comment (sxxgrc) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java:68]: I don't know enough to answer this myself, but I always get concerned when we add a new conditional around central logic. 

What happens to calls to this method that don't satisfy this condition? Do they get an empty config list? How does that differ from what we're doing right now?

My main concern is that we accidentally give empty configs to other services which need to go through the rest of this method.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java [hunk 2]
```diff
@@ -137,9 +143,7 @@ public List<String> getRoleTypes() {
 
     @Override
     public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
-        return !cmTemplateProcessor.isRoleTypePresentInService(HDFS, Lists.newArrayList(NAMENODE))
-                && source.getFileSystemConfigurationView().isPresent()
-                && ConfigUtils.getStorageLocationForServiceProperty(source, CORE_DEFAULTFS).isPresent();
+        return true;
```
Comment (sxxgrc) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java:146]: What is the point of this method now?

**Review by votez — APPROVED** [review]

**Review by romedeiro — APPROVED** [review]

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java [hunk 3]
```diff
@@ -59,39 +59,45 @@ public class CoreConfigProvider extends AbstractRoleConfigProvider {
...
+            apiClusterTemplateConfigs.add(config(HADOOP_RPC_PROTECTION, "privacy"));
         }
 
         return apiClusterTemplateConfigs;
     }
 
     @Override
     public Map<String, ApiClusterTemplateService> getAdditionalServices(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
-        if (isConfigurationNeeded(cmTemplateProcessor, source)
+        if (!cmTemplateProcessor.isRoleTypePresentInService(HDFS, Lists.newArrayList(NAMENODE))
```
Comment (lnardai) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java:98]: This if condition and the previous one on L68 is the same. Please add a meaningful name to it an make a helper method.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java (hunk 2 above)
Comment (lnardai) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java:146]: Why move this condition in to two different parts? This should be the only part where we logically apply a config change. If you change a different config value it should be a different ConfigProvider.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java [hunk 4]
```diff
@@ -81,6 +87,20 @@ protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, Templat
...
+                configs.add(config(DFS_REPLICATION, DFS_REPLICATION_VALUE.toString()));
+            }
+            configs.add(config(DFS_ENCRYPT_DATA_TRANSFER, "true"));
+            configs.add(config(DFS_DATA_TRANSFER_PROTECTION, "privacy"));
+            configs.add(config(HADOOP_RPC_PROTECTION, "privacy"));
+        }
+        return configs;
+    }
+
     private boolean isSDXOptimizationEnabled(TemplatePreparationObject source) {
```
Comment (lnardai) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:104]: We have 2 different version of this logical switch, one here and the other is in the other config provider?
Can't we have the same logic for both?
`entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId()`

**Review by lnardai — CHANGES_REQUESTED** [review]
Please refactor the logic in the configprovider.

**Review by sxxgrc — APPROVED** [review]

**Review by lnardai — APPROVED** [review]

**Pull request merged**